org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/ex/ExternalAntlrLexerFragment.java

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.parser.antlr.ex;
+
+import org.eclipse.xtend2.lib.StringConcatenationClient;
+import org.eclipse.xtext.parser.antlr.Lexer;
+import org.eclipse.xtext.xtext.generator.Issues;
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess;
+import org.eclipse.xtext.xtext.generator.model.IXtextGeneratorFileSystemAccess;
+import org.eclipse.xtext.xtext.generator.model.TypeReference;
+import org.eclipse.xtext.xtext.generator.parser.antlr.AbstractAntlrGeneratorFragment2;
+
+import com.google.common.annotations.Beta;
+
+/**
+ * @since 2.20
+ * @author dietrich - Initial contribution and API
+ */
+@Beta
+public class ExternalAntlrLexerFragment extends AbstractAntlrGeneratorFragment2 {
+	
+	private String lexerGrammar;
+
+	private boolean highlighting;
+
+	private boolean runtime;
+
+	private boolean contentAssist;
+	
+	@Override
+	protected void doGenerate() {
+		// i am not sure when to generate what
+		if (runtime) {
+			StringConcatenationClient binding = new StringConcatenationClient() {
+				@Override
+				protected void appendTo(TargetStringConcatenation target) {
+					target.append("binder.bind(" + Lexer.class.getName() + ".class)"+
+							".annotatedWith(com.google.inject.name.Names.named(" +
+							"org.eclipse.xtext.parser.antlr.LexerBindings.RUNTIME" +
+							")).to(" + lexerGrammar +".class);");
+				}
+			};
+			new GuiceModuleAccess.BindingFactory().addConfiguredBinding("RuntimeLexer", binding).contributeTo(getLanguage().getRuntimeGenModule());
+		}
+		if (highlighting) {
+				if (this.getProjectConfig().getEclipsePlugin().getRoot() != null) {
```

Comment:
maybe put these expressions in some boolean variables e.g. `isEclipsePluginEnabled`? They are repeating quite some times.